chain/syncer.go

```diff
@@ -308,6 +324,53 @@ func (syncer *Syncer) widen(ctx context.Context, ts types.TipSet) (types.TipSet,
 	return wts, nil
 }
 
+// SyncerStatus defines a structured used to represent the state of a syncer.
+type SyncerStatus struct {
+	// Running is true when the syncer is processing ChainInfo, false otherwise.
+	Running bool
+	// NewChain is the ChainInfor the syncer is either activally processing or the last
+	// ChainInfo value the syncer processed.
+	NewChain *types.ChainInfo
+	// Started is the time when the syncer started processing Chain.
+	Started time.Time
+	// Trusted indicates whether or not this sync was triggered by a trusted peer
+	Trusted bool
+	// CurrentHead represents our nodes heaviest verified TipSet
+	CurrentHead types.TipSetKey
+	// CurrentHeight represents the height of our nodes heaviest verified TipSet.
+	CurrentHeight uint64
+}
+
+func (s *SyncerStatus) String() string {
+	return fmt.Sprintf("running: %t, syncing: %s, trusted: %t, started: %s, head: %s, height: %d",
+		s.Running, s.NewChain.String(), s.Started, s.Trusted, s.CurrentHead, s.CurrentHeight)
+}
+
+// Status returns the syncers current status, this includes whether or not the syncer is currently
+// running, the chain its processsing/last processed, and the time it started processing said chain.
+func (syncer *Syncer) Status() *SyncerStatus {
+	return syncer.status
+}
+
+func (syncer *Syncer) startRun(curHead types.TipSetKey, curHeight uint64, trusted bool, newChain *types.ChainInfo) {
```

Comment:
The `start` and `end` run names sound like they do a lot more than they really do. I suggest something like `setRunStatus`.